<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1400</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>上管机图像检测程序</string>
  </property>
  <widget class="QWidget" name="central widget">
   <property name="font">
    <font>
     <pointsize>12</pointsize>
    </font>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <!-- Top row: camera view and logs -->
    <item row="0" column="0">
     <widget class="QGraphicsView" name="graphicsView"/>
    </item>
    <item row="0" column="1">
     <widget class="QListWidget" name="listLogs"/>
    </item>
    <!-- Bottom merged area spanning both columns -->
    <item row="1" column="0" colspan="2">
     <widget class="QGroupBox" name="groupBottom">
      <property name="title">
       <string>Operations</string>
      </property>
      <layout class="QGridLayout" name="gridLayoutBottom">
        <property name="spacing">
         <number>10</number>
        </property>

        <!-- Left sub-area: combined controls -->
        <item row="0" column="0">
          <widget class="QGroupBox" name="groupControls">
            <property name="title"><string>Controls</string></property>
            <property name="minimumHeight"><number>250</number></property>
            <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch><verstretch>0</verstretch>
              </sizepolicy>
            </property>
            <layout class="QVBoxLayout" name="verticalLayoutControls">
              <property name="spacing"><number>5</number></property>

              <!-- Mode selection group -->
              <item>
                <widget class="QGroupBox" name="groupMode">
                  <property name="title"><string>Mode</string></property>
                  <layout class="QGridLayout" name="gridLayoutModes">
                    <property name="horizontalSpacing"><number>4</number></property>
                    <property name="verticalSpacing"><number>4</number></property>
                    <item row="0" column="0">
                      <widget class="QComboBox" name="comboMode">
                        <property name="sizePolicy">
                          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                            <horstretch>0</horstretch><verstretch>0</verstretch>
                          </sizepolicy>
                        </property>
                      </widget>
                    </item>
                  </layout>
                </widget>
              </item>
              <!-- Save image checkbox group -->
              <item>
                <widget class="QGroupBox" name="groupSave">
                  <property name="title"><string>Save Options</string></property>
                  <layout class="QVBoxLayout" name="verticalLayoutSave">
                    <property name="spacing"><number>5</number></property>

                    <item><widget class="QCheckBox" name="chkSaveImage"><property name="text"><string>保存图片</string></property></widget></item>
                  </layout>
                </widget>
              </item>
              <!-- Camera controls group -->
              <item>
                <widget class="QGroupBox" name="groupCamera">
                  <property name="title"><string>Camera</string></property>
                  <layout class="QHBoxLayout" name="horizontalLayoutCamera">
                    <property name="spacing"><number>20</number></property>

                    <item><widget class="QPushButton" name="btnConnectCamera"><property name="text"><string>连接相机</string></property><property name="minimumWidth"><number>120</number></property><property name="minimumHeight"><number>40</number></property><property name="maximumWidth"><number>120</number></property><property name="maximumHeight"><number>40</number></property></widget></item>
                    <item><widget class="QLabel" name="labelCameraStatus"><property name="text"><string>相机状态： 尚未连接</string></property></widget></item>
                  </layout>
                </widget>
              </item>
            </layout>
          </widget>
        </item>
        <!-- Right sub-area: configuration parameters -->
        <item row="0" column="1">
          <widget class="QGroupBox" name="groupConfig">
            <property name="title"><string>Config</string></property>
            <property name="sizePolicy"><sizepolicy hsizetype="Preferred" vsizetype="Preferred"><horstretch>0</horstretch><verstretch>0</verstretch></sizepolicy></property>
            <layout class="QVBoxLayout" name="verticalLayoutConfig">
                <!-- —— “提示区”，初始隐藏 —— -->
                <item>
                  <widget class="QLabel" name="labelConfigWarning">
                    <property name="text">
                      <string>修改 Camera 或 Plc 参数后，请务必重新连接相机</string>
                    </property>
                    <property name="styleSheet">
                      <string>color: red; font-weight: bold; font-size:22px;</string>
                    </property>
                    <property name="visible">
                      <bool>false</bool>  <!-- 默认隐藏 -->
                    </property>
                  </widget>
                </item>
                <!-- —— 参数区 —— -->
                <item>
                  <widget class="QWidget" name="configContent">
                    <layout class="QGridLayout" name="gridConfig">
                      <!-- Dynamic config widgets will be added here at runtime -->
                    </layout>
                  </widget>
                </item>
           </layout>
          </widget>
        </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
